wp-content/civi-extensions/goonjcustom/Civi/DroppingCenterService.php

```diff
@@ -217,6 +335,87 @@ public static function setOfficeDetails($op, $groupID, $entityID, &$params) {
 
   }
 
+  /**
+   *
+   */
+  public static function processDispatchEmail(string $op, string $objectName, int $objectId, &$objectRef) {
+
+    if ($objectName !== 'AfformSubmission') {
+      return;
+    }
+
+    $afformName = $objectRef->afform_name;
+
+    if ($afformName !== 'afformSendDispatchEmail') {
+      return;
+    }
+
+    $jsonData = $objectRef->data;
+    $dataArray = json_decode($jsonData, TRUE);
+
+    $droppingCenterId = $dataArray['Eck_Collection_Camp1'][0]['fields']['id'] ?? NULL;
+
+    if (!$droppingCenterId) {
+      return;
+    }
+
+    $droppingCenterData = EckEntity::get('Collection_Camp', TRUE)
+      ->addSelect('Collection_Camp_Core_Details.Contact_Id', 'Dropping_Centre.Goonj_Office')
+      ->addWhere('id', '=', $droppingCenterId)
+      ->execute()->single();
+
+    $contactId = $droppingCenterData['Collection_Camp_Core_Details.Contact_Id'] ?? NULL;
+    $goonjOffice = $droppingCenterData['Dropping_Centre.Goonj_Office'] ?? 'N/A';
+
+    if (!$contactId) {
+      return;
+    }
+
+    $contactInfo = Contact::get(TRUE)
+      ->addSelect('email_primary.email', 'phone_primary.phone', 'display_name')
+      ->addWhere('id', '=', $contactId)
+      ->execute()->single();
+
+    $email = $contactInfo['email_primary.email'];
+    $phone = $contactInfo['phone_primary.phone'];
+    $initiatorName = $contactInfo['display_name'];
+
+    // Send the dispatch email.
+    self::sendDispatchEmail($email, $initiatorName, $droppingCenterId, $contactId, $goonjOffice);
+  }
+
+  /**
+   *
+   */
+  public static function sendDispatchEmail($email, $initiatorName, $droppingCenterId, $contactId, $goonjOffice) {
+    $homeUrl = \CRM_Utils_System::baseCMSURL();
+    $from = HelperService::getDefaultFromEmail();
+
+    $vehicleDispatchFormUrl = $homeUrl . '/vehicle-dispatch/#?Camp_Vehicle_Dispatch.Collection_Camp=' . $droppingCenterId . '&Camp_Vehicle_Dispatch.Filled_by=' . $contactId . '&Camp_Vehicle_Dispatch.To_which_PU_Center_material_is_being_sent=' . $goonjOffice . '&Eck_Collection_Camp1=' . $droppingCenterId;
+
+    $emailHtml = "
+    <html>
+    <body>
+    <p>Dear {$initiatorName},</p>
+    <p>Thank you so much for your invaluable efforts in running the Goonj Dropping Center. 
+    Your dedication plays a crucial role in our work, and we deeply appreciate your continued support.</p>
+    <p>Please fill out this Dispatch Form – <a href='{$vehicleDispatchFormUrl}'>[link]</a> once the vehicle is loaded and ready to head to Goonj’s processing center. 
+    This will help us to verify and acknowledge the materials as soon as they arrive.</p>
+    <p>We truly appreciate your cooperation and continued commitment to our cause.</p>
+    <p>Warm Regards,<br>Team Goonj..</p>
+    </body>
+    </html>
+    ";
+    $mailParams = [
+      'subject' => 'Kindly fill the Dispatch Form for Material Pickup',
```

Comment:
do we need to put dropping center code and address here?

Reply:
No, as per requirement , we don't want dropping center code or address.